models/issues/review.go

```diff
@@ -618,9 +635,33 @@ func AddReviewRequest(ctx context.Context, issue *Issue, reviewer, doer *user_mo
 		return nil, err
 	}
 
-	// skip it when reviewer hase been request to review
-	if review != nil && review.Type == ReviewTypeRequest {
-		return nil, committer.Commit() // still commit the transaction, or committer.Close() will rollback it, even if it's a reused transaction.
+	if review != nil {
+		// NOTE: in case of failure we still commit the transaction here, else committer.Close() will rollback it,
+		// even if it's a reused transaction.
+
+		// skip it when reviewer hase been request to review
+		if review.Type == ReviewTypeRequest {
+			return nil, committer.Commit()
+		}
+
+		if issue.IsClosed {
+			var err error = ErrReviewRequestOnClosedPR{}
```

Comment:
It's very weird. It will succeed but users will get a failure information?

Reply:
Hmm, I'm not quite sure what you mean by succeeds. To me I see this as:

- The user is trying to perform an add review request
- The reviewer has already made a review before (which means to me that this is a re-review request)
- The PR is closed, so we should return the error `ErrReviewRequestOnClosedPR{}` to callers

But now that I'm reading the code again I think I might see what you mean. Does the problem have to do with the transaction being committed that I do when I return a error?

Reply:
I mean we should not Commit the transaction, just let it rollback.

Reply:
Gotcha. Being this low I wasn't too sure on how to handle database transactions and wrongly assumed from reading the existing comment here that I should commit no matter if it was a success or failure. Will make the necessary changes.

Reply:
Fixed